ecl/hql/hqlfold.cpp

```diff
@@ -2154,6 +2154,74 @@ IHqlExpression * foldConstantOperator(IHqlExpression * expr, unsigned foldOption
             }
             break;
         }
+    case no_regex_findset:
+        {
+            IValue * t0 = expr->queryChild(0)->queryValue();
+            IValue * t1 = expr->queryChild(1)->queryValue();
+            if (t0 && t1)
+            {
+                if(isUnicodeType(t0->queryType()))
+                {
+                    unsigned plen = t0->queryType()->getStringLen();
+                    OwnedPtr<UChar> pattern (new UChar[plen+1]);
+                    t0->getUCharStringValue(plen+1, pattern.get()); //plen+1 so get null-terminated
+
+                    unsigned slen = t1->queryType()->getStringLen();
+                    OwnedPtr<UChar> search (new UChar[slen]);
+                    t1->getUCharStringValue(slen, search);
+
+                    ICompiledUStrRegExpr * compiled = rtlCreateCompiledUStrRegExpr(pattern.get(), !expr->hasAttribute(noCaseAtom));
+                    assertex(compiled!=NULL);
+
+                    HqlExprArray results;
+                    IUStrRegExprFindInstance * match = compiled->find(search, 0, slen);
+                    while (match->found())
```

Comment:
@ghalliday  by code-generator help object, do you mean ICompiledUStrRegExpr *match?  That class was always been stateful.

When an error occurs in rtlCreateCompiledUstr, an exception is thrown which does seem to be handled well... for example:
Warning:  (5,1): error C0000: Bad regular expression: Found a closing ) with no corresponding openening parenthesis.  The error occurred while parsing the regular expression fragment: 'a:]]|(-))+>>>HERE>>>)|([[:digi'.: ([:alpha:]]|(-))+)|([[:digit:]]+([.][[:digit:]]+)?) (0, 0), 0,